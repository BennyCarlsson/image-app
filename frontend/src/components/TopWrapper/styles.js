import styled from 'styled-components'

export const StyledTopWrapper = styled.div`
  display: flex;
  background-color: #ffe8e5;
  width: 100%;
  height: 500px;
  border-radius: 0 0 50% 50% / 0 0 30% 30%;
  transform: scaleX(1.3);
  overflow: hidden;
  box-shadow: 0px 2px 17px -11px rgb(96 96 96);
  margin-bottom: 24px;
`
export const Inner = styled.span`
  transform: scaleX(calc(1 / 1.3));
  width: 100%;
  height: 100%;
`
