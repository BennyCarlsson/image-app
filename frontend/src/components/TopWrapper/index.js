import React from 'react'
import { Inner, StyledTopWrapper } from './styles'
const TopWrapper = ({ children }) => {
  return (
    <StyledTopWrapper>
      <Inner>{children}</Inner>
    </StyledTopWrapper>
  )
}
export default TopWrapper
