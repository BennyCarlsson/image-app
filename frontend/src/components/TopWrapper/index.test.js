import React from 'react'
import { render, screen } from '@testing-library/react'
import TopWrapper from '.'

describe('<TopWrapper />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<TopWrapper />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<TopWrapper />)
    expect(screen).toBeTruthy()
  })

  it('should render child', () => {
    render(
      <TopWrapper>
        <p>I'm TopWrapper child</p>
      </TopWrapper>
    )
    expect(screen.getByText("I'm TopWrapper child")).toBeInTheDocument()
  })
})
