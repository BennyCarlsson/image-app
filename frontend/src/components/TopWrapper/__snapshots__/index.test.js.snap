// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopWrapper /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL isRMAQ"
  >
    <span
      class="sc-gsDKAQ kcwFXz"
    />
  </div>
</DocumentFragment>
`;
